File: Strings/dictionarySort.py
def dictionarySort(s):
    count = [0] * 26

    for i in range(len(s)):
        index = ord(s[i]) - ord("a")
        count[index] += 1

    res = []
    for i in range(26):
        for j in range(count[i]):
            res.append(chr(ord("a") + i))

    return "".join(res)


ans = dictionarySort("bhrwiolpabbrrwwhlhp")
print(ans)

File: Strings/DictionarySort.java
public class DictionarySort {
    String dictionarySort(String s) {
        int[] count = new int[26];
        for (int i = 0; i < s.length(); i++) {
            int index = s.charAt(i) - 'a';
            count[index]++;
        }

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < 26; i++) {
            for (int j = 0; j < count[i]; j++) {
                sb.append((char) ('a' + i));
            }
        }

        return sb.toString();
    }

    public static void main(String[] args) {
        String s = "bhrwiolpabbrrwwhlhp";

        System.out.println(new DictionarySort().dictionarySort(s));
    }
}
